service/src/main/java/bio/terra/cbas/controllers/PublicApiController.java

```diff
@@ -31,10 +31,14 @@ public ResponseEntity<SystemStatus> getStatus() {
       URL url = new URL(this.cromwellConfig.healthUri());
       URLConnection connection = url.openConnection();
       connection.setConnectTimeout(5000);
-      var stream = connection.getInputStream();
-
-      result = new String(stream.readAllBytes(), StandardCharsets.UTF_8);
-      isOk = true;
+      connection.setReadTimeout(5000);
+      try (var stream = connection.getInputStream()) {
+        result = new String(stream.readAllBytes());
+        isOk = true;
+      } catch (Exception e) {
+        result = e.getLocalizedMessage();
+        isOk = false;
+      }
```

Comment:
You don't need to duplicate the `catch` here, you can have a `try` with no `catch` or `finally`
```suggestion
      }
```

Reply:
Oh, so the uncaught exception will bubble up to the outer `try`?

Reply:
Yes, only a matching `catch` clause will stop the exception propagation